### ISSUE TITLE
[BUG] [1.16.5] Issues with the Cook/Restaurant

### ISSUE THREAD

**Issue by Sandriell** [ldtteam/minecolonies#8231]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.16

### MineColonies Version

1.16.5-1.0.580-ALPHA

### Structurize Version

1.16.5-1.0.371-ALPHA

### Related Mods and their Versions

https://gist.github.com/Sandriell/0688b4e8e61c46e1bd44c36ad9ff6fc0

### Current Behavior

Two issues with the Cook/Assistant Cook...

1. Similar to the [previously reported issue](https://github.com/ldtteam/minecolonies/issues/8103) with the Assistant Cook keeping what he makes, the cook is doing the same thing- resulting in endless loops with minimum stock requirements not being met. I watched him make several stacks before interfering and moving one of them to the hut's rack and stopping the loop. Though thinking about this maybe the real issue is that the cook is making the food in the first place, and the Assistant Cook should be doing it? These are taught recipes as well, not simple cooked food.
2.  They fail to keep track of ingredients they already have. I am not sure exactly where in the process this is breaking down, but the inventory for the Restaurant will fill up with ingredients over time that the cook completely ignores, thus requesting even more. This is the issue that completely breaks food from mods like Pam's Harvestcraft because they fail to see the utensils they already have.

### Expected Behavior

Cook puts what he makes in the inventory, or what is in his inventory counts towards the minimum stock since he is the one handing out the food anyway.

Using the ingredients that are already in the hut inventory first before requesting more.

### Reproduction Steps

1. Teach the restaurant a recipe.
2. Set a minimum stock for that item.
3. Wait for/supply all the ingredients - sometimes my manually moving them back to the warehouse so they are "seen" again.
4. Cook cook makes the request
5. Immediately requests to make another stock to try and satisfy the minimum stock requirement

### Logs

N/A

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by MotionlessTrain** [comment]
Is this a smelting recipe or a 3x3 crafting recipe?

**Comment by Raycoms** [comment]
Not sure I'd call this a bug, a minimum stock is the min not the max stock. So, the cook might make more than the min actually.

**Comment by Sandriell** [comment]
It is a 3x3 crafting recipe.

Well yes a minimum is just that, but I would say the potential endless loop is a bug. It was 3 completed stacks before I interfered and that was only because I was out of the needed ingredients anyway.

Part 2 is really the bigger issue in my opinion.

**Comment by Raycoms** [comment]
Cooks shouldn't do any 3x3 recipes at all

**Comment by Sandriell** [comment]
Yes, why I added the part about "Though thinking about this maybe the real issue is that the cook is making the food in the first place, and the Assistant Cook should be doing it? These are taught recipes as well, not simple cooked food."

But I watched the cook make it. The Assistant cook was off wandering around and the ingredients disappeared and instantly appeared in the Cooks inventory as the stack of food.
